.github/workflows/ci.yml

```diff
@@ -267,3 +267,40 @@ jobs:
         run: |
           docker build -f ./Dockerfile_linter \
             --build-arg DEPENDENCY_IMAGE=ghcr.io/genspectrum/lapis-silo-dependencies:commit-${{ env.HEAD_SHA }}-amd64 . 
+    
+  unitTests:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Generate dependency files hash
+        run: |
+          DIR_HASH=$(echo -n ${{ hashFiles('conanfile.py', 'conanprofile.docker', './Dockerfile_dependencies') }})
+          echo "DIR_HASH=$DIR_HASH" >> $GITHUB_ENV
+
+      - name: Choose docker image for executing unit tests
+        run: |
+          DEPENDENCY_TAG="filehash-${{ env.DIR_HASH }}-amd64"
+
+          # Dependency image could already have the cached up-to-date dependencies, :latest is only rebuilt once a day
+          DEPENDENCY_IMAGE="${{ env.DOCKER_DEPENDENCY_IMAGE_NAME }}:${DEPENDENCY_TAG}"
+          LATEST_IMAGE="${{ env.DOCKER_DEPENDENCY_IMAGE_NAME }}:latest"
+
+          if docker manifest inspect ${DEPENDENCY_IMAGE} > /dev/null 2>&1; then
+            echo "Using image: ${DEPENDENCY_IMAGE}"
+            DOCKER_IMAGE=${DEPENDENCY_IMAGE}
+          else
+            echo "Image ${IMAGE_FULL} not found, falling back to :latest"
+            DOCKER_IMAGE=${LATEST_IMAGE}
+          fi
+          
+          echo "DOCKER_IMAGE=$DOCKER_IMAGE" >> $GITHUB_ENV
+      
+      - name: Run CI in Docker
+        run: |
+          docker run --rm \
+            -v ${{ github.workspace }}:/workspace \
+            -w /workspace \
+            ${{ env.DOCKER_IMAGE }} \
+            bash -login -c "mv /src/conanprofile . && make test"
```

Comment:
There is already a "Run unit tests" step somewhere above. Do we need both of them?